spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -567,7 +567,32 @@
         )
       end
 
-      context 'when USPS returns a 4xx status code' do
+      context 'when USPS returns an unexpected 400 status code' do
+        before(:each) do
+          stub_request_proofing_results_with_responses(
+            {
+              status: 400,
+              body: { 'responseMessage' => 'This USPS location has closed 😭' }.to_json,
+            },
+          )
+        end
+
+        it_behaves_like(
+          'enrollment_encountering_an_exception',
+          exception_class: 'Faraday::BadRequestError',
+          exception_message: 'the server responded with status 400',
+          response_message: 'This USPS location has closed 😭',
+          response_status_code: 400,
+        )
+
+        it 'logs the error to NewRelic' do
+          expect(NewRelic::Agent).to receive(:notice_error).
+            with(instance_of(Faraday::BadRequestError))
+          job.perform(Time.zone.now)
+        end
```

Comment:
Would we want "logs the error to NewRelic" to be an assertion as part of the shared example set `enrollment_encountering_an_exception` ?

Reply:
Yeah that makes sense. The only downside is it's a little complicated to make the example set know which type of error `NewRelic::Agent#receive` should be called with so I lean towards just expecting the method to be called and not specifying the error type it should receive

Reply:
Actually there are a couple of error types which we aren't currently reporting to NewRelic, such as if USPS returns something to us other than a hash when we expect a hash. I don't know if it's useful to log those to NewRelic...I guess we could construct a StandardError with some error string content

I'm now leaning towards leaving it out of the example set, how it is right now